app/gui/src/project-view/components/ComponentBrowser/input.ts

```diff
@@ -168,7 +176,20 @@ export function useComponentBrowserInput(
     newText: string
     requiredImport: ProjectPath | undefined
   } {
-    if (sourceNodeIdentifier.value) {
+    const displayOwner = (owner: ProjectPath) => {
+      if (owner.path) return qnLastSegment(owner.path)
+      if (owner.project) return qnLastSegment(owner.project)
+      return 'Main' as Ast.Identifier
+    }
```

Comment:
Is it not a method of `ProjectPath`? If not, I still think it could be. It doesn't seem to be only about displaying - in the code we want to put `Main`s as well.

Reply:
Added some helpers and refactored this code.